common/src/main/java/com/graphicsfuzz/common/util/SideEffectChecker.java

```diff
@@ -45,6 +48,17 @@ public void visitFunctionCallExpr(FunctionCallExpr functionCallExpr) {
           // Assume that any call to a non-builtin might have a side-effect.
           predicateHolds();
         }
+        for (FunctionPrototype p :
+            TyperHelper.getBuiltins(shadingLanguageVersion).get(functionCallExpr.getCallee())) {
+          // We check each argument of the built-in's prototypes to see if they require lvalues -
+          // if so, they can cause side effects.
```

Comment:
Also add a note to say that we could try to be more precise by resolving exactly which overload of the builtin is being invoked.